hydromt_wflow/workflows/demand.py

```diff
@@ -0,0 +1,839 @@
+    swfrac.attrs = {"_FillValue": -9999}
+    swfrac = swfrac.copy()
+
+    # Set and interpolate the values
+    swfrac.values[x, y] = gwfrac_val
+    if interpolate:
+        swfrac = swfrac.interpolate_na(dim=swfrac.raster.x_dim, method="linear")
+        swfrac = swfrac.interpolate_na(
+            dim=swfrac.raster.x_dim, method="linear", fill_value="extrapolate"
+        )
+
+    # Set the nodata values based on the dem of the model (da_like)
+    swfrac = swfrac.where(da_like != da_like.raster.nodata, -9999)
+
+    # Return surface water frac
+    return swfrac
+
+
+def classify_pixels(
+    da_crop: xr.DataArray,
+    da_model: xr.DataArray,
+    threshold: float,
+    nodata_value: float | int = -9999,
+):
+    """Classifies pixels based on a (fractional) threshold.
+
+    Pixels with a value above this threshold are set to be irrigated, and pixels below
+    this value as rainfed.
+
+    Parameters
+    ----------
+    da_crop: xr.DataArray
+        Data with crop mask
+    da_model: xr.DataArray
+        Layer of the wflow model (used for reprojecting)
+    threshold: float
+        Threshold above which pixels are classified as irrigated
+    nodata_value: float | int = -9999
+        Value to be used for nodata
+
+    Returns
+    -------
+    crop_map: xr.DataArray
+        Mask with classified pixels
+    """
+    # Convert crop map to an area map
+    da_area = da_crop * da_crop.raster.area_grid()
+    # Resample to model grid and sum areas
+    da_area2model = da_area.raster.reproject_like(da_model, method="sum")
+
+    # Calculate relative area
+    relative_area = da_area2model / da_area2model.raster.area_grid()
+
+    # Classify pixels with a 1 where it exceeds the threshold, and 0 where it doesnt.
+    crop_map = xr.where(relative_area >= threshold, 1, 0)
+    crop_map = xr.where(da_model.isnull(), nodata_value, crop_map)
+
+    # Fix nodata values
+    crop_map.raster.set_nodata(nodata_value)
+
+    return crop_map
+
+
+def find_paddy(
+    landuse_da: xr.DataArray,
+    irrigated_area: xr.DataArray,
+    paddy_class: int,
+    nodata_value: float | int = -9999,
+):
+    """Find the location of paddy crops.
+
+    Done by finding the overlap between the irrigated area
+    grid, and a landcover map that has a separate class for rice/paddy fields.
+
+    Parameters
+    ----------
+    landuse_da: xr.DataArray
+        Map with landuse classes (must contain a class for paddy)
+    irrigated_area: xr.DataArray
+        Mask with irrigated areas
+    paddy_class: int
+        Class that matches the rice or paddy land cover type
+    nodata_value: float | int = -9999
+        Value to be used as nodata value
+
+    Returns
+    -------
+    paddy: xr.DataArray
+        Mask with pixels classified as paddy or rice fields
+    nonpaddy: xr.DataArray
+        Maks with pixels classified as non paddy irrigation
+    """
+    # Resample irrigated area to landuse datasets
+    irr2lu = irrigated_area.raster.reproject_like(landuse_da)
```

Comment:
Maybe do specify here the resampling method (nearest I guess), in case HydroMT decides to change the default in the future for any reason. Also makes the code more readable here.